File: apps/api-go/internal/response/response.go
package response

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type Body struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data"`
	TraceID string      `json:"traceId"`
}

func OK(c *gin.Context, data interface{}) {
	c.JSON(http.StatusOK, Body{Code: 0, Message: "success", Data: data, TraceID: traceID(c)})
}

func Created(c *gin.Context, data interface{}) {
	c.JSON(http.StatusCreated, Body{Code: 0, Message: "success", Data: data, TraceID: traceID(c)})
}

func Fail(c *gin.Context, status int, code int, message string) {
	c.JSON(status, Body{Code: code, Message: message, Data: gin.H{}, TraceID: traceID(c)})
}

func traceID(c *gin.Context) string {
	if value := c.GetHeader("X-Trace-Id"); value != "" {
		return value
	}
	return c.GetString("traceId")
}

File: apps/ai-service-python/app/core/response.py
from typing import Any

from fastapi.responses import JSONResponse


def ok(data: Any, trace_id: str = "") -> dict[str, Any]:
    return {"code": 0, "message": "success", "data": data, "traceId": trace_id}


def fail(status_code: int, code: int, message: str, trace_id: str = "") -> JSONResponse:
    return JSONResponse(
        status_code=status_code,
        content={"code": code, "message": message, "data": {}, "traceId": trace_id},
    )

